lib/phail_web/live/phail.ex

```diff
@@ -91,11 +91,10 @@ defmodule PhailWeb.Live.Phail do
   end
 
   defp change_labels(conversation_id, old_label, new_label) do
+    conversation = Conversation.get(conversation_id)
     if old_label != nil do
```

Comment:
don't really like this `if` here. there might be a better way to refactor this, maybe with pattern-matching in the function signature.

also you could combine both removing an adding into the same db transaction if you want. [Ecto.Multi](https://hexdocs.pm/ecto/Ecto.Multi.html)

Reply:
Any particular reason to avoid `if` here? I don't see a refactor that doesn't end up less clear.

Reply:
(I'll definitely look into Ecto.Multi later, thanks.)

Reply:
mostly me not liking `if` in functional stuff. XD i also don't see a possible refactor at this time, so can be parked for later

`Ecto.Multi` is an awesome awesome thing! another use case i can think of is in denormalisation. eg: if you wanna put the latest conversation time on the convo table so that you don't have to max message time. you'd probably wanna update `conversation.latest_message_time` when you add a new message to a conversation.

Reply:
The if syntax is really annoying in erlang which led to me writing a bunch of `maybe_do_x` functions, which ended being a bit of a readability nightmare. I'm kind of glad of elixirs nicer if syntax.

Looks like that bit is performing well atm. Triggers are really useful for denormalisation too.